src/Orc.Automation.Recorder/ViewModels/MainViewModel.cs

```diff
@@ -1,155 +1,164 @@
+        BuildTree(informer);
 
-            RaiseCanExecute();
-        }
+        RaiseCanExecute();
+    }
 
-        private bool CanStart()
-        {
-            return !_inProcess && SelectedCandidateHandler is not null;
-        }
+    private bool CanStart()
+    {
+        return !_inProcess && SelectedCandidateHandler is not null;
+    }
 
-        private void OnPause()
-        {
-            _inProcess = false;
+    private void BuildTree(AutomationInformer informer)
+    {
+        var service = new TreeBuilderService();
+        service.Build(informer);
+    }
 
-            RaiseCanExecute();
-        }
+    private void OnPause()
+    {
+        _inProcess = false;
 
-        private bool CanPause()
-        {
-            return _inProcess;
-        }
+        RaiseCanExecute();
+    }
 
-        private void OnStop()
-        {
-            _inProcess = false;
+    private bool CanPause()
+    {
+        return _inProcess;
+    }
 
-            var selectedCandidateHandler = SelectedCandidateHandler;
-            if (selectedCandidateHandler is null)
-            {
-                return;
-            }
+    private void OnStop()
+    {
+        _inProcess = false;
 
-            var handler = new IntPtr(selectedCandidateHandler.Value);
+        var selectedCandidateHandler = SelectedCandidateHandler;
+        if (selectedCandidateHandler is null)
+        {
+            return;
+        }
 
-            var window = AutomationElement.FromHandle(handler);
-            var informer = window.Find<AutomationInformer>();
+        var handler = new IntPtr(selectedCandidateHandler.Value);
 
-            informer.StopRecord();
+        var window = AutomationElement.FromHandle(handler);
+        var informer = window.Find<AutomationInformer>();
 
-            RaiseCanExecute();
-        }
+        informer.StopRecord();
 
-        private bool CanStop()
-        {
-            return _inProcess;
-        }
+        RaiseCanExecute();
+    }
 
-        private void OnComment()
-        {
+    private bool CanStop()
+    {
+        return _inProcess;
+    }
+
+    private void OnComment()
+    {
             
-        }
+    }
 
-        private bool CanComment()
-        {
-            return !string.IsNullOrWhiteSpace(CommentText);
-        }
+    private bool CanComment()
+    {
+        return !string.IsNullOrWhiteSpace(CommentText);
+    }
 
-        private void OnSend()
-        {
+    private void OnSend()
+    {
             
-        }
+    }
 
-        private bool CanSend()
-        {
-            return !string.IsNullOrWhiteSpace(Message);
-        }
+    private bool CanSend()
+    {
+        return !string.IsNullOrWhiteSpace(Message);
+    }
 
-        private void RaiseCanExecute()
-        {
-            Start.RaiseCanExecuteChanged();
-            Stop.RaiseCanExecuteChanged();
-            Pause.RaiseCanExecuteChanged();
-        }
+    private void RaiseCanExecute()
+    {
+        Start.RaiseCanExecuteChanged();
```

Comment:
Is this sufficient? Or should all commands be re-evaluated?

Reply:
![image](https://github.com/WildGums/Orc.Automation/assets/5163834/e2533993-a407-485f-89f1-1bb89ecd21c8)